### PR TITLE
Update snapshot documentation with storage_type option

### LINKED ISSUE #1684: [DOC] Searchable Snapshot Feature
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.**
 We would like to develop the documentation for the initial experimental release of the searchable snapshot feature in OpenSearch 2.4. The feature design proposal is here: https://github.com/opensearch-project/OpenSearch/issues/3895

**What other resources are available?**

The searchable snapshot feature consists primarily of two API changes:

- A new parameter is added to the snapshot restore API indicating the snapshot should be restored as a "searchable snapshot" as opposed to downloading all index data at restore time
- A new node role is added to indicate that a node is capable of hosting a searchable snapshot index and serving queries.

I've included some thoughts on the documentation about these two areas below:

### Snapshot Restore API

A new parameter will be introduced in the [snapshot restore API](https://opensearch.org/docs/latest/opensearch/snapshot-restore/#restore-snapshots): `storage_type`.

| Setting | Description|
| --- | --- |
| storage_type | Must be one of `local` or `remote_snapshot`. `local` is the default if not specified, and indicates that all snapshot metadata and index data will be downloaded to local instance storage. `remote_snapshot` indicates that snapshot metadata will be downloaded to the cluster but the remote repository will remain the authoritative store of the index data. Data will be downloaded on-demand as necessary to service queries. At least one node in the cluster must be configured for the `search` role in order to restore a snapshot of type `remote_snapshot`. |

### Node Role

This feature adds a new `search` role to the list of possible node roles. I can't seem to find existing documentation about all the existing build-in roles (`data`, `cluster_manager`, `ingest`, and `remote_cluster_client`), but here is a example documentation on a [dynamic role added by a plugin](https://opensearch.org/docs/latest/ml-commons-plugin/index/#ml-node).

### PR DESCRIPTION
**Pull request by ariamarble**
Signed-off-by: ariamarble <armarble@amazon.com>

### Description
New option update

### Issues Resolved
#1684 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/snapshots/snapshot-restore.md [hunk 1]
```diff
@@ -343,6 +343,7 @@ Setting | Description
 `rename_replacement` | If you want to rename indices as you restore them, use this option to specify the replacement pattern. Use `$0` to include the entire matching index name, `$1` to include the content of the first capture group, etc.
 `index_settings` | If you want to change index settings on restore, specify them here.
 `ignore_index_settings` | Rather than explicitly specifying new settings with `index_settings`, you can ignore certain index settings in the snapshot and use the cluster defaults on restore.
+`storage_type` | `local` indicates that all snapshot metadata and index data will be downloaded to local storage. <br /><br >`remote_snapshot` indicates that snapshot metadata will be downloaded to the cluster, but the remote repository will remain the authoritative store of the index data. Data will be downloaded and cached as necessary to service queries. At least one node in the cluster must be configured with the [search role]({{site.url}}{{site.baseurl}}/security-plugin/access-control/users-roles/) in order to restore a snapshot using the type `remote_snapshot`. <br /><br >Defaults to `local`.
```
Comment (Naarcha-AWS) [inline _opensearch/snapshots/snapshot-restore.md:346]: Just adding a space between one line break.

```suggestion
`storage_type` | `local` indicates that all snapshot metadata and index data will be downloaded to local storage. <br /><br > `remote_snapshot` indicates that snapshot metadata will be downloaded to the cluster, but the remote repository will remain the authoritative store of the index data. Data will be downloaded and cached as necessary to service queries. At least one node in the cluster must be configured with the [search role]({{site.url}}{{site.baseurl}}/security-plugin/access-control/users-roles/) in order to restore a snapshot using the type `remote_snapshot`. <br /><br > Defaults to `local`.
```

**Review by Naarcha-AWS — APPROVED** [review]

**Review by kolchfa-aws — APPROVED** [review]
LGTM

File: _opensearch/snapshots/snapshot-restore.md [hunk 2]
```diff
@@ -343,6 +343,7 @@ Setting | Description
 `rename_replacement` | If you want to rename indices as you restore them, use this option to specify the replacement pattern. Use `$0` to include the entire matching index name, `$1` to include the content of the first capture group, etc.
 `index_settings` | If you want to change index settings on restore, specify them here.
 `ignore_index_settings` | Rather than explicitly specifying new settings with `index_settings`, you can ignore certain index settings in the snapshot and use the cluster defaults on restore.
+`storage_type` | `local` indicates that all snapshot metadata and index data will be downloaded to local storage. <br /><br > `remote_snapshot` indicates that snapshot metadata will be downloaded to the cluster, but the remote repository will remain the authoritative store of the index data. Data will be downloaded and cached as necessary to service queries. At least one node in the cluster must be configured with the [search role]({{site.url}}{{site.baseurl}}/security-plugin/access-control/users-roles/) in order to restore a snapshot using the type `remote_snapshot`. <br /><br > Defaults to `local`.
```
Comment (andrross) [inline _opensearch/snapshots/snapshot-restore.md:346]: Roles [in the security plugin](https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/) are a different concept. The only existing documentation I've found for configuring the node roles is in the [ML plugin](https://opensearch.org/docs/latest/ml-commons-plugin/index/#ml-node). I haven't found any documentation around the built-in node roles (cluster_manager, data, ingest, etc) that would be appropriate to link here.

**Review by andrross — COMMENTED** [review]
This feature will be experimental in 2.4, similar to [segment replication](https://opensearch.org/docs/latest/opensearch/segment-replication/index/).

We'll want the same caveats here, as well as instructions for enabling the feature flag. I don't know if it's right to have this parameter added to the main documentation for snapshot restore while it is experimental, given the hoops a user has to jump through to enable it.

**Pull request merged**